src/main/java/net/sf/rails/algorithms/NetworkTrain.java

```diff
@@ -63,10 +63,10 @@ public static NetworkTrain createFromString(String trainString) {
         boolean ignoreTowns = false; int multiplyCities = 1; int multiplyTowns = 1;
         boolean isHTrain = false;
         boolean isETrain = false;
-        if (t.equals("D")) {
+        if ( "D".equals(t)) {
```

Comment:
A note for a future changes:
These strings look like constants to me that should be places in `public final static` Strings

Reply:
A lot of these might even be better represented by a `switch`